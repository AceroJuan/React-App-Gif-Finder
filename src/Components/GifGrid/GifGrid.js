import React from "react";
import useGifs from "../../utils/hooks/useGifs";
import GifGridItem from "../GifGridItem/GifGridItem";
import PropTypes from "prop-types";

const GifGrid = ({ category }) => {
  const { data: images, loading } = useGifs(category);

  return (
    <>
      {loading && (
        <p className="animate__animated animate__flash">Cargando ... </p>
      )}
      {category ? (
        <h2 className="card-grid__h1">Resultados para {category}</h2>
      ) : null}
      <div className="card-grid">
        {images?.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
