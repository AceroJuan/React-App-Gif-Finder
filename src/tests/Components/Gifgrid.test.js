import React from "react";
import "@testing-library/jest-dom";
import GifGrid from "../../Components/GifGrid/GifGrid";
import { shallow } from "enzyme";
import useGifs from "../../utils/hooks/useGifs";
jest.mock("../../utils/hooks/useGifs.js");

describe("Pruebas en <GifGrid />", () => {
  const category = "hola mundo";

  test("debe mostrar <GifGrid /> de manera correcta", () => {
    useGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar items cuando se cargan imagenes usando useGifs", () => {
    const images = [
      {
        id: "ABC",
        url: "https://localhost:3000/cualquier/cosa.jpg",
        title: "Cualquier cosa",
      },
      {
        id: "123",
        url: "https://localhost:3000/cualquier/cosa.jpg",
        title: "Cualquier cosa",
      },
    ];
    useGifs.mockReturnValue({
      data: images,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    // expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(images.length);
  });
});
