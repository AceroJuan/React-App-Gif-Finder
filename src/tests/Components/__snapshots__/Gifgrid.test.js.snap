// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe mostrar <GifGrid /> de manera correcta 1`] = `
<Fragment>
  <p
    className="animate__animated animate__flash"
  >
    Cargando ... 
  </p>
  <h2
    className="card-grid__h1"
  >
    Resultados para 
    hola mundo
  </h2>
  <div
    className="card-grid"
  />
</Fragment>
`;
